File: python/전화번호-목록.py
def solution(phone_book):
    
    phone_book.sort()
    
    for i in range(len(phone_book) - 1):
        if len(phone_book[i]) < len(phone_book[i+1]) and phone_book[i+1].startswith(phone_book[i]):
            return False
        
    return True

File: java/전화번호-목록.java
import java.util.*;

class Solution {
    // <전화번호, 인덱스>
    private static Map<String, Integer> phone_head = new HashMap<>();

    public boolean solution(String[] phone_book) {
        
    // 정렬
    Arrays.sort(phone_book);
        
    // map 생성
    for(int i=0; i<phone_book.length; i++){
        phone_head.put(phone_book[i], i);
    }
        
    // 접두어 존재 여부 탐색
    for(int i=0; i<phone_book.length; i++){
        for(int j=0; j<phone_book[i].length(); j++){
            if(phone_head.containsKey(phone_book[i].substring(0, j))) return false;
        }
    }
        return true;
    }
}
